arch/arc/include/swap_macros.h

```diff
@@ -121,9 +142,30 @@
 	ld_s r13, [sp, ___callee_saved_stack_t_dpfp2h_OFFSET]
 	sr r13, [_ARC_V2_FPU_DPFP2H]
 #endif
-2 :
 #endif
-
+3 :
+#ifdef CONFIG_DSP_SHARING
+	ld_s r13, [r2, ___thread_base_t_user_options_OFFSET]
+	/* K_DSP_REGS is bit 5 */
+	bbit0 r13, 5, 4f
```

Comment:
This will break after someone touched the bit definitions in `include/zephyr/kernel.h`
Can we somehow use `K_DSP_REGS` directly here instead of hardcoding the value (5)?

Reply:
bit(5) is from thread option and I added two functions to set and reset it. Not sure how to configure different threads without using thread option. I think using  `K_DSP_REGS`  macro might affect all threads.